### PR TITLE
docs(api.md): add info for .$x() methods

### PR DESCRIPTION
**Pull request by vsemozhetbyt**
Possibly connected issue: https://github.com/puppeteer/puppeteer/issues/6710

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1217,7 +1217,7 @@ Shortcut for [page.mainFrame().$eval(selector, pageFunction)](#frameevalselector
 - `expression` <[string]> Expression to [evaluate](https://developer.mozilla.org/en-US/docs/Web/API/Document/evaluate).
 - returns: <[Promise]<[Array]<[ElementHandle]>>>
 
-The method evaluates the XPath expression.
+The method evaluates the XPath expression relative to the page document as its context node. If there are no such elements, the method will resolve to an empty array.
```
Comment (mathiasbynens) [inline docs/api.md]:
```suggestion
The method evaluates the XPath expression relative to the page document as its context node. If there are no such elements, the method resolves to an empty array.
```

File: docs/api.md [hunk 2]
```diff
@@ -2889,7 +2889,7 @@ const html = await frame.$eval('.main-container', e => e.outerHTML);
 - `expression` <[string]> Expression to [evaluate](https://developer.mozilla.org/en-US/docs/Web/API/Document/evaluate).
 - returns: <[Promise]<[Array]<[ElementHandle]>>>
 
-The method evaluates the XPath expression.
+The method evaluates the XPath expression relative to the frame document as its context node. If there are no such elements, the method will resolve to an empty array.
```
Comment (mathiasbynens) [inline docs/api.md]:
```suggestion
The method evaluates the XPath expression relative to the frame document as its context node. If there are no such elements, the method resolves to an empty array.
```

File: docs/api.md [hunk 3]
```diff
@@ -3545,7 +3545,13 @@ expect(await tweetHandle.$eval('.retweets', node => node.innerText)).toBe('10');
 - `expression` <[string]> Expression to [evaluate](https://developer.mozilla.org/en-US/docs/Web/API/Document/evaluate).
 - returns: <[Promise]<[Array]<[ElementHandle]>>>
 
-The method evaluates the XPath expression relative to the elementHandle. If there are no such elements, the method will resolve to an empty array.
+The method evaluates the XPath expression relative to the `elementHandle` as its context node. If there are no such elements, the method will resolve to an empty array.
```
Comment (mathiasbynens) [inline docs/api.md]:
```suggestion
The method evaluates the XPath expression relative to the `elementHandle` as its context node. If there are no such elements, the method resolves to an empty array.
```

**Comment by vsemozhetbyt** [comment]
@mathiasbynens Thank you, the commit is amended according to your suggestions (and an additional correction was added to sync wording).

**Review by mathiasbynens — APPROVED** [review]

**Review by mathiasbynens — APPROVED** [review]

**Pull request merged**